tests/vm_test.go

```diff
@@ -122,6 +122,13 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			Expect(reviewResponse.Details.Causes).To(HaveLen(1))
 			Expect(reviewResponse.Details.Causes[0].Field).To(Equal("spec.template.spec.domain.devices.disks[2].name"))
 		})
+		It("should reject creation if template spec does not contain any memory settings", func() {
+			vmi := libvmi.NewCirros()
+			vmi.Spec.Domain.Resources.Requests = nil
+			vm := tests.NewRandomVirtualMachine(vmi, false)
+			_, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm)
+			Expect(err).To(MatchError("admission webhook \"virtualmachine-validator.kubevirt.io\" denied the request: no memory requested, at least one of 'spec.template.spec.domain.memory.guest', 'spec.template.spec.domain.memory.hugepages.size' or 'spec.template.spec.domain.resources.requests.memory' must be set"))
```

Comment:
nit - Any chance of breaking this down into it's component parts to make the assertions cleaner?